### DISCUSSION TITLE
How to define the routes using different method

### DISCUSSION THREAD

**Discussion by BinNong** [fastapi/fastapi#7631] [Questions] [answered]
### Description

How can I define the route supporting 'get' and 'post' methos simutaneously? As like the codes in Flask: "@app.route('/api/test', methods=['GET', 'POST'])"

**Comment by jorgerpo** [comment]
Can you show the function in your  flask code?

**Comment by Toad2186** [comment, marked as answer]
You can do something like this:
```
@app.get('/route')
@app.post('/route')
async def myhandler():
    pass
```

But each of the HTTP verbs have different semantics, so in the general case you probably should have different handlers...

**Comment by jorgerpo** [comment]
exactly

**Comment by BinNong** [comment]
thanks

**Comment by tiangolo** [comment]
Thanks for the help here eveyone! :bowing_man: 

Thanks @BinNong for reporting back and closing the issue :+1:
